### DISCUSSION TITLE
Status flag if driver running but no browser connected

### DISCUSSION THREAD

**Discussion by dornech** [GCuser99/SeleniumVBA#119] [Ideas]
For some purpose it would be helpful to be able to differentiate if the driver has an open session assigned or not.
This is implicitly stored in Webdriver.session_ but sessioID_ it is not available outside the webdriver object.
Simple function checking session_id <> vbnullstring would fix this.

**Comment by GCuser99** [comment]
Hi @dornech - we do have a way to get that and various other information about the session via a Dictionary object:
```vba
'prints the session id string - returns "no open browser session exists" if not yet assigned
Debug.Print driver.GetSessionsInfo()("id") 
```
Will that be sufficient for your use case?

**Comment by dornech** [comment]
Had a look into it - in principle yes - but only Chrome if I am right this time ?

**Comment by GCuser99** [comment]
Chrome, Edge, and IE. Not Firefox.

**Comment by dornech** [comment]
I need to check this browser type-independent but I think I found the solution

```
' check status WebDdriver
Function checkWebdriverRunnning(ByRef objWebDriver As SeleniumVBA.webdriver) As Boolean

   Dim objDict As New Dictionary

   On Error Resume Next
   Set objDict = objWebDriver.GetDriverStatus
   checkWebdriverRunnning = (objDict.Count > 0)
   On Error GoTo 0
   
End Function
   

' check status WebDdriver
Function checkBrowserOpened(ByRef objWebDriver As SeleniumVBA.webdriver) As Boolean

   Dim s_UserAgent As String

   On Error Resume Next
   s_UserAgent = objWebDriver.GeTuseRagent
   checkBrowserOpened = (Len(s_UserAgent) > 0)
   On Error GoTo 0
   
End Function
```
